dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs

```diff
@@ -65,7 +90,30 @@ public KernelArguments(IDictionary<string, object?> source, Dictionary<string, P
     /// <summary>
     /// Gets or sets the prompt execution settings.
     /// </summary>
-    public IReadOnlyDictionary<string, PromptExecutionSettings>? ExecutionSettings { get; set; }
+    /// <remarks>
+    /// The settings dictionary is keyed by the service ID, or <see cref="PromptExecutionSettings.DefaultServiceId"/> for the default execution settings.
+    /// When setting, the service id of each <see cref="PromptExecutionSettings"/> must match the key in the dictionary.
+    /// </remarks>
+    public IReadOnlyDictionary<string, PromptExecutionSettings>? ExecutionSettings
+    {
+        get => this._executionSettings;
+        set
+        {
+            if (this._executionSettings is { Count: > 0 })
+            {
+                foreach (var kv in this._executionSettings!)
```

Comment:
Should this be performed on incoming `value` rather than existing `this._executionSettings`? Looks like it's the case in `PromptTemplateConfig.cs`:
```suggestion
            if (value is { Count: > 0 })
            {
                foreach (var kv in value!)
```